src/SqlClient/SqlClientProvider.fs

```diff
@@ -29,6 +29,19 @@ type SqlProgrammabilityProvider(config : TypeProviderConfig) as this =
 
     let cache = new MemoryCache(name = this.GetType().Name)
     let methodsCache = new MemoryCache(name = this.GetType().Name)
+    
+    let splitKeyVal (s : string) = 
+        match s.Split(':') with 
+        | [|key; value|] -> (key.Trim().ToLower(), value)
+        |_ -> invalidArg "s" "parameter must be of type key:value"
+    
+    let mapFromString (s : string) =
```

Comment:
I'd rename this `mapFromString` to something more descriptive of the usage of this code ("why/what") rather than "how" it is done, for example `parseResultSetsDeclarationList`, and then make `splitKeyVal` local to that function.

Reply:
I don't know if passing the string is the correct way, I'd prefer to pass a Map to as the static parameter but I couldn't find how. Maybe is a constrain in what type can be passed. I'm not really happy passing the string with the dictionary syntax. 

Also I'm not sure if the syntax I use is the best, I was considering also: 
key1->value1, key2->value2.... but again, this is not relevant, what is really important is to pass the information to the TP so it knows when to append to the sys.sp_describe_first_result_set command. 

If we ended up using this with a string this code maybe even needs to be moved to a better place, this was just a way to keep the change small while proving that something like this might work.

Reply:
For now AFAIK, the only things that can be passed to a TP as static parameters are literals (string, numbers and enum), we are indeed stuck to string and figuring out a format that makes the most sense.

Workarounds for now:

* make a sql command as I was describing, basically containing the same text you are running through `sys.sp_describe_first_result_set`
* changing the stored procedures involving temp tables to run the code as dynamic SQL and add the `with result sets` inside

I'm doing the later in most cases but I understand it makes maintenance of those procedures a more annoying.